sys/xtimer/xtimer_core.c

```diff
@@ -23,32 +23,42 @@
 #include "irq.h"
 
 /* WARNING! enabling this will have side effects and can lead to timer underflows. */
-#define ENABLE_DEBUG 0
+#define ENABLE_DEBUG (0)
 #include "debug.h"
 
-static volatile int _in_handler = 0;
+static int _in_handler = 0;
 
 static volatile uint32_t _long_cnt = 0;
+
+static uint32_t _safe_long_cnt(uint32_t now)
+{
+    uint32_t long_value = _long_cnt;
+    return (long_value >> 1) + ((long_value & 1) && (~now >> 31));
+}
+
 #if XTIMER_MASK
 volatile uint32_t _high_cnt = 0;
```

Comment:
Since you are doing already some refactoring and  it's not static... what about renaming it to `_xtimer_high_cnt`? :-)